packager/packager.cc

```diff
@@ -23,6 +23,27 @@ namespace {
 constexpr string_view PACKAGE_FILE_NAME = "__package.rb"sv;
 constexpr core::NameRef TEST_NAME = core::Names::Constants::Test();
 
+bool isPackageModule(const core::GlobalState &gs, core::SymbolRef modName) {
+    while (modName.exists() && modName != core::Symbols::root()) {
+        if (modName == core::Symbols::PackageRegistry() || modName == core::Symbols::PackageTests()) {
+            return true;
+        }
+        modName = modName.data(gs)->owner;
+    }
+    return false;
+}
+
+class PrunePackageModules final {
+public:
+    ast::ExpressionPtr postTransformClassDef(core::Context ctx, ast::ExpressionPtr tree) {
```

Comment:
If you did this as `preTransformClassDef` then you would avoid traversing the class... but you wouldn't be able to return an `EmptyTree` since `preTransformClassDef` expects that you return a `ClassDef`:

https://github.com/sorbet/sorbet/blob/5279843c/ast/treemap/treemap.h#L168

idk if it would actually make things faster, but... you could hollow out the class in `preTransformClassDef` and return `EmptyTree` in `postTransformClassDef`.

Reply:
I added a pre-step to clear the rhs. Also had to store a stack of "was this a package module" to avoid doing the check more than once. This made it slightly faster.